testdata/testing.go

```diff
@@ -33,14 +35,26 @@ var benchmarks = []testing.InternalBenchmark{}
 
 var examples = []testing.InternalExample{}
 
+var errMain = errors.New("testing: unexpected use of func Main")
+
+// matchStringOnly is part of upstream, and is used below to provide a dummy deps to pass to MainStart
+// so it can be run with go (tested with go 1.16) to provide a baseline for the regression test.
+// See c56cc9b3b57276.  Unfortunately, testdeps is internal, so we can't just use &testdeps.TestDeps{}.
```

Comment:
> Unfortunately, testdeps is internal, so we can't just use &testdeps.TestDeps{}.
	
That's not the reason: the testing package can most certainly import internal packages because they're both part of GOROOT. The reason that testdeps exists is so that the testing package can import useful packages while those packages can still import the testing package (for their own tests) without creating a circular dependency.

Reply:
Oops, nevermind. The comment is correct. I didn't realize the code was in testdata/testing.go (instead of src/testing.go).